src/Upscale.cpp

```diff
@@ -104,6 +104,7 @@ void Upscale::SolveCoarse(const mfem::BlockVector& x, mfem::BlockVector& y) cons
 
     coarse_solver_->Solve(x, y);
     y *= -1.0;
+    OrthogonalizeCoarse(y);
```

Comment:
This (lines 106 and 107) should probably be added to the non-block `SolveCoarse` also (right after line 90).